### PR TITLE
Fix regression in Angular ICU expression

### PR DESCRIPTION
**Pull request by seiyab**
## Description
Issue: #16914 

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] ~(If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).~
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/parser-html.js [hunk 1]
```diff
@@ -90,6 +91,12 @@ function normalizeAngularIcuExpression(node) {
   if (node.type === "plural" || node.type === "select") {
     node.clause = node.type;
     node.type = "angularIcuExpression";
+    for (const c of node.cases) {
+      const lastExpr = c.expression?.at(-1);
+      if (lastExpr && lastExpr.type === "text") {
+        lastExpr.value = htmlWhitespaceUtils.trimEnd(lastExpr.value);
+      }
+    }
```
Comment (seiyab) [inline src/language-html/parser-html.js:99]: I'm not familiar with Angular. I'm not sure enough that this is valid change.

**Comment by fisker** [comment]
Hi @seiyab, thanks for working on this, I didn't mean to ask you to make changes to the HTML printer, #16922 should be a better fix, I mentioned you to confirm if the output is expected without this change, because I feel there maybe something wrong when printing "fill".

**Comment by fisker** [comment]
In another word, is this expected?

**Prettier 3.4.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBtAOlABFg5AMwglwBpMcLKsBzAJwgFcAHACg2ys4EsoATBGG3KcRAGx5wyHEVXQgYtBnDlSZnOVmArhainUat2u7nwFDpxygGcI+GOKiSdlivi6jR5lyLkAZCLRwALbaFt44Dk5h4XJcTFYMISCq4ZSRKak4crwQogGhmRTpzqlyVlwwBYVYxdHecgCGQXAwZMkl4bXVWHKQUFZwYPAwDLRVhV3VjbxxXFZgPNTjmZOFcnDircupq5llcLzbnRIZeyA5R967pSBwXAxWQRCH7XUu1zHywUyX7ycd3lQAF1Tt4gQBKUHGCFQ3Q2OwfGQwgE+EAAX1+nERVGRbxwEMwQOSIAgTBgXGgVmQoAatHoAHcAAq0hBUlANUT0hoATypJBAACNaA0wABrFoAZSacF8EmQ+A5A35QpF4pgEqYIsWyAUSn5AyCXB1ikktwAHkw4LQuM1YByACpWqC0rhwNkK0RKkDlKDUURwACKDAg8HlitNACsrGaJYt-UGQ3Aw57TQBHYPwRn0eLIEANKwAWkcBwOxIUDXciwAwhAgkEGrmOaJiT6-XAAIIwBRcAUMTNW2WOZNegAWMCCogA6iOKm7NWA4BLWRUuAA3Crc3NgKx8kCrpQASVMsAlYGtZPbfAlMG5-uHpqY9AGk+FTFzj7dVtXSf5PAGtBgLMGmoet735TVaH-XMcjAAs4AtPJAloYlAnTLhAiAkCGyQD0vRgBoBUnLheBgEdkAAJn5B44HtAi2V1U1ggFA5+F4XwGl9BhgLgAAxAJ6y7bV2T7CB0TRIA)

```sh
--parser doc-explorer
--use-tabs
```

**Input:**

```jsx
[
  'foo',
          group([
            indent([
              line,
              "true",
              " {",
              group([
                indent([
                  softline,
                  fill([
                    "Lorem",
                    line,
                    "ipsum",
                    line,
                    "dolor",
                    line,
                    "sit",
                    line,
                    "amet,",
                    line,
                    "consectetur",
                    line,
                    "adipiscing",
                    line,
                    "elit,",
                    line,
                    "sed",
                    line,
                    "do",
                    line,
                    "eiusmod",
                    line,
                    "temp",
                    line,
                    [],
                  ]),
                ]),
                softline,
              ]),
              "}",
              line,
            ]),
          ])
]
```

**Output:**

```jsx
foo
	true {
		Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod temp

	}
	
```

**Comment by seiyab** [comment]
Ah, I was hasty. I'll close this.
I think the output is technically correct at least. The last word "temp" fits in the line but the next doc `line` doesn't. `line` breaks as a result.
I'll check whether older printer also reproduce it or not later.

**Comment by seiyab** [comment]
Prettier main + reverting 94fa86e1642fbec2bbb154ee8c55bc9399 also reproduces https://github.com/prettier/prettier/issues/16914.
It is not caused by the new printer but reproducible with the older printer.

**Comment by seiyab** [comment]
Probably Prettier omits trailing whitespace so removing broken`line` followed immediately by broken `line` by printer might make sense but doesn't look straightforward.

**Comment by fisker** [comment]
Good to know, thank you for your time.

> omits trailing whitespace

This is actually bug, but I don't think we can change the behavior, it will break too many things.

**Pull request closed without merging**